src/stack/PostMessageTransport.js

```diff
@@ -81,7 +81,7 @@ easyXDM.stack.PostMessageTransport = function(config){
         // #ifdef debug
         trace("received message '" + event.data + "' from " + origin);
         // #endif
-        if (origin == targetOrigin && event.data.substring(0, config.channel.length + 1) == config.channel + " ") {
+        if (origin == targetOrigin && (typeof event.data === 'string' || event.data instanceof String) && event.data.substring(0, config.channel.length + 1) == config.channel + " ") {

```

Comment:
The only legit type for `event.data` in this context is `string`, and not the boxed String type - let's simplify this.